lib/regexp/regexp.go

```diff
@@ -0,0 +1,290 @@
+	"find_all":        starlark.NewBuiltin("find_all", findAll),
+	"find_submatches": starlark.NewBuiltin("find_submatches", findSubmatches),
+	"matches":         starlark.NewBuiltin("matches", matches),
+	"replace_all":     starlark.NewBuiltin("replace_all", replaceAll),
+	"split":           starlark.NewBuiltin("split", split),
+}
+
+func builtinAttr(recv starlark.Value, name string, methods map[string]*starlark.Builtin) (starlark.Value, error) {
+	b := methods[name]
+	if b == nil {
+		return nil, nil // no such method
+	}
+	return b.BindReceiver(recv), nil
+}
+
+func builtinAttrNames(methods map[string]*starlark.Builtin) []string {
+	names := make([]string, 0, len(methods))
+	for name := range methods {
+		names = append(names, name)
+	}
+	sort.Strings(names)
+	return names
+}
+
+func matches(_ *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var src string
+
+	if err := starlark.UnpackPositionalArgs(b.Name(), args, kwargs, 1, &src); err != nil {
+		return nil, err
+	}
+
+	re := b.Receiver().(*Regexp).re
+	return starlark.Bool(re.MatchString(src)), nil
+}
+
+func find(_ *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var src string
+
+	if err := starlark.UnpackPositionalArgs(b.Name(), args, kwargs, 1, &src); err != nil {
+		return nil, err
+	}
+
+	re := b.Receiver().(*Regexp).re
+	return starlark.String(re.FindString(src)), nil
+}
+
+func findAll(_ *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var (
+		src string
+		max int = -1
+	)
+
+	if err := starlark.UnpackPositionalArgs(b.Name(), args, kwargs, 1, &src, &max); err != nil {
+		return nil, err
+	}
+
+	re := b.Receiver().(*Regexp).re
+	return toList(re.FindAllString(src, max)), nil
+}
+
+func findSubmatches(_ *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var src string
+
+	if err := starlark.UnpackPositionalArgs(b.Name(), args, kwargs, 1, &src); err != nil {
+		return nil, err
+	}
+
+	re := b.Receiver().(*Regexp).re
+	return toList(re.FindStringSubmatch(src)), nil
+}
+
+func replaceAll(thread *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var (
+		src  string
+		repl starlark.Value
+	)
+
+	if err := starlark.UnpackPositionalArgs(b.Name(), args, kwargs, 2, &src, &repl); err != nil {
+		return nil, err
+	}
+
+	re := b.Receiver().(*Regexp).re
+	switch x := repl.(type) {
+	case starlark.Callable:
+		var fnErr error
+		fn := func(matched string) string {
+			res, err := starlark.Call(thread, repl, starlark.Tuple{starlark.String(matched)}, nil)
+			if err != nil {
+				// Save the error to be able to return it to the caller
+				fnErr = fmt.Errorf("%s: An error occured while executing the callable: %s", x.Name(), err.Error())
```

Comment:
fnErr = err // propagate error unmodified, with its call stack

Let the call stack do the talking.